src/formElementPlugins/dropdownChoice/src/ApplicationView.tsx

```diff
@@ -30,43 +28,46 @@ const ApplicationView: React.FC<ApplicationViewProps> = ({
     options,
     optionsDisplayProperty,
     hasOther,
-    default: defaultOption,
+    default: defaultValue,
+    replaceResponseOnDefaultChange,
   } = parameters as {
     label?: string
     description?: string
     placeholder?: string
     search?: boolean
     hasOther?: boolean
-    default: any
+    default?: string | number
+    replaceResponseOnDefaultChange?: boolean
   } & (ObjectOptions | StringOptions)
 
-  const [selectedIndex, setSelectedIndex] = useState<number>()
+  const [selectedIndex, setSelectedIndex] = useState<number | undefined>(
+    currentResponse?.optionIndex ?? undefined
+  )
   const [addedOption, setAddedOption] = useState<string | null>(null)
 
   const { isEditable } = element
 
   useEffect(() => {
-    // This ensures that, if a default is specified, it gets saved on first load
-    if (!currentResponse?.text && defaultOption !== undefined) {
-      const optionIndex = getDefaultIndex(defaultOption, options)
-      onSave({
-        text:
-          optionsDisplayProperty &&
-          options[optionIndex] !== undefined &&
-          typeof options[optionIndex] === 'object'
-            ? (options[optionIndex] as ObjectOption)[optionsDisplayProperty]
-            : options[optionIndex],
-        selection: options[optionIndex],
-        optionIndex,
-        isCustomOption: false,
-      })
-      setSelectedIndex(optionIndex)
-    }
-    if (currentResponse?.text) {
-      const { optionIndex } = currentResponse
-      setSelectedIndex(optionIndex)
+    // This deals with the case when a default response has been externally set
+    // (e.g. by a ListBuilder) and only a selection (with no index) has been
+    // provided.
+    if (options[0] === 'Loading...') return
+    if (currentResponse?.selection || currentResponse?.optionIndex === undefined) {
+      setSelectedIndex(getDefaultIndex(currentResponse?.selection, options))
```

Comment:
Hmmm wondering where this case has gone?
```
if (currentResponse?.text) {
      const { optionIndex } = currentResponse
      setSelectedIndex(optionIndex)
}
```

Reply:
That's now handled by the `useDefault` hook:
```
if (replaceResponseOnDefaultChange || !currentResponse?.text) onChange(defaultValue)
```

:)